LoRaEngine/LoraKeysManagerFacade/FCntCacheCheck.cs

```diff
@@ -42,41 +43,63 @@ public FCntCacheCheck(ILoRaDeviceCacheStore deviceCache)
 
             EUIValidator.ValidateDevEUI(devEUI);
 
+            if (!uint.TryParse(fCntUp, out uint clientFCntUp))
+            {
+                string errorMsg = "Missing FCntUp";
+                throw new ArgumentException(errorMsg);
+            }
+
             if (!string.IsNullOrEmpty(abpFcntCacheReset))
             {
-                this.deviceCache.KeyDelete(devEUI);
+                using (var deviceCache = new LoRaDeviceCache(this.deviceCache, devEUI, gatewayId))
+                {
+                    if (await deviceCache.TryToLockAsync())
+                    {
+                        if (deviceCache.TryGetInfo(out var deviceInfo))
+                        {
+                            // only reset the cache if the current valud is larger
```

Comment:
Small typo on comment `valud` &rarr; `value`

`continued processing` &rarr; `continue processing`

Reply:
continued is right here. continued refers to the initial resetter and that carried on (continued) processing and has added it again.